pyocd/core/coresight_target.py

```diff
@@ -86,8 +86,9 @@ def elf(self, filename):
             self._elf = None
         else:
             self._elf = ELFBinaryFile(filename, self.memory_map)
-            self.cores[0].elf = self._elf
-            self.cores[0].set_target_context(ElfReaderContext(self.cores[0].get_target_context(), self._elf))
+        for core_number in range(len(self.cores)):
```

Comment:
The reason it was only applied to the first core is that each core normally has different code. What's really needed is support for an ELF file for each core. But I'm happy to include this change as a temporary workaround.

Reply:
Yes, I thought about that as well. One elf file per core. In my case, the elf file is shared.

Reply:
I think there are use cases for both. Shared and separated elf file. It depends on how the memory for the device was implemented. I will keep an eye out for the non-shared elf file situation. I wish the gdb server interface provided a symbol address as well as the size. I think I will ask for an enhancement with gdb but I doubt they will do it as it might break things